cmd/container-disk-v2alpha/main_test.go

```diff
@@ -16,6 +17,7 @@ import (
 var containerDiskBinary string
 
 func init() {
+	testing.Init()
```

Comment:
This is not a recommended fix. The problem is the call to flag.Parse which is wrong.

flag.Parse Is called by the generated test main, all others should just register the flags.

Reply:
First of all, sorry for not replying, had lots of emails from GitHub(with fail tests).

I am not author of this commit but I try my best to answer.
I believe this "fix" was based on [this](https://golang.org/doc/go1.13#testing). It was intended to keep it function as before as this is not the goal of this PR. 

I see this as a specific case when we want to parse the args to be able to modify it. 
I can see that this can be achieved in BeforeEach as follow:
`if !strings.Contains(containerDiskBinary, ../../){
   	containerDiskBinary = filepath.Join("../../", containerDiskBinary)
}`

I fail to see why this is a prefered way to do this. Can you explain what negatives does this brings?

Reply:
This line will explicitly call the parsing on pre-main which is wrong as it will mask all other flags registrations from other places.
The way flags work is that each library/module/package registers flags they support and in main, once, you call parse in order to process them all. If you call them before, other flags from other sources are not included and even though parse is called again in the test main, it will have no affect (parse is a singleton, secondary calls have no meaning).

Bottom line, there is no need to call this explicitly here, it will be called by the generated test main.

Reply:
Ref: https://github.com/kubevirt/kubevirt/pull/3786

Reply:
I don't see that `Init`calls parsing. Also, I don't see that `flags.Parse()` is singleton (
// Parse parses the command-line flags from os.Args[1:]. Must be called
// after all flags are defined and before flags are accessed by the program.).

I think that we would have failing tests if there was some masking of flags. (This would be the negative impact)
I will align 2 test which left to be aligned with #3786.

Reply:
> I don't see that Init calls parsing.

Right, it just registers the test framework flags and is called from main. More reason not to call it here, as it is called from the generated main.

> I don't see that flags.Parse() is singleton

Right, it is the testing.Init who is a singleton and in general the test main will not call `Parse` if it was already called.

The main point is: "Must be called after all flags are defined and before flags are accessed", so you cannot call parse before main, it should be called only at main (after all other inits have been processed).